android/src/main/java/com/tailscale/ipn/ui/service/IpnManager.kt

```diff
@@ -4,27 +4,26 @@
 
 package com.tailscale.ipn.ui.service
 
-import android.util.Log
 import com.tailscale.ipn.ui.localapi.LocalApiClient
 import com.tailscale.ipn.ui.notifier.Notifier
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.cancel
 
 class IpnManager {
-    var notifier = Notifier()
-    var apiClient = LocalApiClient()
-    val model: IpnModel
-    
-    constructor() {
-        model = IpnModel(notifier, apiClient)   
-    }
+    private var notifier = Notifier()
+    private var scope = CoroutineScope(SupervisorJob() + Dispatchers.Main)
```

Comment:
This doesn't quite get us to LifecycleScope, but I think it's an improvement to just have a single scope that we pass in to the other types that need it.

I'm not actually convinced about the current structure of using a singleton IpnManager and IpnModel, but I'm not ready to make any concrete suggestions on that.